lib/api/apiUtils/bucket/bucketDeletion.js

```diff
@@ -111,15 +111,19 @@ export function deleteBucket(bucketMD, bucketName, canonicalID, log, cb) {
 
     return async.waterfall([
         function checkForObjectsStep(next) {
-            return metadata.listObject(bucketName, { maxKeys: 1 }, log,
-                (err, objectsListRes) => {
+            const params = { maxKeys: 1, listingType: 'DelimiterVersions' };
```

Comment:
so, this will return all versions even the null version if versioning has not been enabled?  please add comment in the code